pallets/thea-executor/src/lib.rs

```diff
@@ -504,24 +521,32 @@ pub mod pallet {
 					polkadex_primitives::AssetId::Asset(deposit.asset_id),
 					polkadex_primitives::AssetId::Polkadex
 				];
-
+				let amount_out: T::AssetBalanceAdapter = T::ExistentialDeposit::get().into();
+				let asset1: T::MultiAssetIdAdapter =
+					polkadex_primitives::AssetId::Asset(deposit.asset_id).into();
+				let asset2: T::MultiAssetIdAdapter = polkadex_primitives::AssetId::Polkadex.into();
+				let fee_amount: T::AssetBalanceAdapter = pallet_asset_conversion::pallet::Pallet::<T>
+				::quote_price_tokens_for_exact_tokens(asset1.into(), asset2.into(), amount_out.into(), true)
+					.ok_or(Error::<T>::CannotSwapForFees)?.into();
+				let fee_amount: u128 = fee_amount.into();
+				ensure!(fee_amount <= deposit_amount, Error::<T>::AmountCannotBeZero);
+				Self::resolve_mint(&Self::thea_account(), deposit.asset_id.into(), fee_amount)?;
+				T::Swap::swap_tokens_for_exact_tokens(
```

Comment:
Use the return value instead of fee_amount to get the correct value to deduct

Reply:
Does it matter? 
Follwing code will also return exact value
```
pallet_asset_conversion::pallet::Pallet::<T>
				::quote_price_tokens_for_exact_tokens(asset1.into(), asset2.into(), amount_out.into(), true)
					.ok_or(Error::<T>::CannotSwapForFees)?.into();
```
Maybe I can add ensure statement to make sure both values are same.